### PR TITLE
Added a few new features and fixed a few existing features

### PR DESCRIPTION
**Pull request by yofreke**
developed alongside the saucy trucks game, matches master of entities

### CODE REVIEW

File: src/ui/ButtonView.js [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    }
+
+    this.normalImage = opts.image;
+    this.pressedImage = opts.pressedImage;
+    this.disabledImage = opts.disabledImage;
+
+    this.toggle = opts.toggle || false;
+
+    this._btnID = uid++;
+    this._pubsub = new PubSub();
```
Comment (rogueSkib) [inline src/ui/ButtonView.js]: View's inherit from PubSub, aka Emitter, doesn't seem like adding an additional instance of PubSub is necessary

File: src/group/index.js [hunk 2]
```diff
@@ -30,6 +55,16 @@ exports = {
...
 
   /**
+   * Automatically obtain a group for this class, and return an actor instance from that group.
+   * This function provides an easy way to add custom actors to scene.
+   * @func scene.addCustomActor
+   * @arg     {Actor}  ctor
+   * @arg     {object} [opts]
+   * @returns {Actor}  newInstance
+   */
+  addCustomActor: addCustomActor,
```
Comment (rogueSkib) [inline src/group/index.js]: Why is this fn defined outside of the exports object?

Comment (yofreke) [inline src/group/index.js]: Not sure - we should decide what the contribution guidelines are for scene...  longer discussion required

Comment (yofreke) [inline src/group/index.js]: will fix this as i go through on the most recent doc audit

**Pull request merged**